core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/AiServicesProcessor.java

```diff
@@ -1120,8 +1120,16 @@ public void handleAiServices(
                                 ignoredPredicates,
                                 tools, toolQualifierProviderItems);
                         if (!methodCreateInfo.getToolClassInfo().isEmpty()) {
-                            unremovableBeanProducer.produce(UnremovableBeanBuildItem
-                                    .beanClassNames(methodCreateInfo.getToolClassInfo().keySet().toArray(EMPTY_STRING_ARRAY)));
+                            methodCreateInfo.getToolClassInfo().keySet().stream()
+                                    .map(toolClassName -> {
+                                        try {
+                                            return Thread.currentThread().getContextClassLoader().loadClass(toolClassName);
+                                        } catch (ClassNotFoundException e) {
+                                            throw new IllegalStateException(e);
+                                        }
+                                    })
```

Comment:
Do we really need to load the class?

Reply:
I also don't like this. Alternatively I'd need the `DotName` of the class. I tried to carry it inside the  `AiServiceMethodCreateInfo` instead of the `String` as it does now, only to realize that I cannot do that because that class is in the runtime which doesn't depend on jandex. Is there a way to recreate the `DotName` of the class here starting from the `String` class name?

Reply:
`DotName.createSimple(toolClassName);` is what you are looking for

Reply:
Thanks for suggestion, fixed.